web/src/pages/api/public/media/[mediaId].ts

```diff
@@ -0,0 +1,123 @@
+import { z } from "zod";
+
+import { env } from "@/src/env.mjs";
+import { getMediaStorageServiceClient } from "@/src/features/media/server/getMediaStorageClient";
+import {
+  GetMediaQuerySchema,
+  GetMediaResponseSchema,
+  PatchMediaBodySchema,
+} from "@/src/features/media/validation";
+import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
+import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
+import {
+  ForbiddenError,
+  InternalServerError,
+  LangfuseNotFoundError,
+} from "@langfuse/shared";
+import { Prisma, prisma } from "@langfuse/shared/src/db";
+import { recordIncrement, recordHistogram } from "@langfuse/shared/src/server";
+
+export default withMiddlewares({
+  GET: createAuthedAPIRoute({
+    name: "Get Media data",
+    querySchema: GetMediaQuerySchema,
+    responseSchema: GetMediaResponseSchema,
+    fn: async ({ query, auth }) => {
+      if (auth.scope.accessLevel !== "all") throw new ForbiddenError();
+
+      const { projectId } = auth.scope;
+      const { mediaId } = query;
+
+      const media = await prisma.media.findFirst({
+        where: {
+          projectId,
+          id: mediaId,
+        },
+      });
+
+      if (!media) throw new LangfuseNotFoundError("Media asset not found");
+      if (!media.uploadHttpStatus)
+        throw new LangfuseNotFoundError("Media not yet uploaded");
+      if (media.uploadHttpStatus !== 200)
+        throw new LangfuseNotFoundError(`Media upload failed`);
+
+      const mediaStorageClient = getMediaStorageServiceClient(media.bucketName);
+      const ttlSeconds = env.LANGFUSE_S3_MEDIA_DOWNLOAD_URL_EXPIRY_SECONDS;
+      const urlExpiry = new Date(Date.now() + ttlSeconds * 1000).toISOString();
+
+      const url = await mediaStorageClient.getSignedUrl(
+        media.bucketPath,
+        ttlSeconds,
+        false,
+      );
+
+      return {
+        mediaId,
+        contentType: media.contentType,
+        contentLength: Number(media.contentLength),
+        url,
+        urlExpiry,
+      };
+    },
+  }),
+
+  PATCH: createAuthedAPIRoute({
+    name: "Update Media Uploaded At",
+    querySchema: z.object({
+      mediaId: z.string(),
+    }),
+    bodySchema: PatchMediaBodySchema,
+    responseSchema: z.void(),
+    fn: async ({ query, body, auth }) => {
+      if (auth.scope.accessLevel !== "all") throw new ForbiddenError();
+
+      const { projectId } = auth.scope;
+      const { mediaId } = query;
+      const { uploadedAt, uploadHttpStatus, uploadHttpError, uploadTimeMs } =
+        body;
+
+      try {
+        await prisma.media.update({
+          where: {
+            projectId,
+            id: mediaId,
+          },
+          data: {
+            uploadedAt,
+            uploadHttpStatus,
+            uploadHttpError: uploadHttpStatus === 200 ? null : uploadHttpError,
+          },
+        });
+
+        recordIncrement("media.upload_http_status", 1, {
```

Comment:
can you prefix all metrics with `langfuse.`? This helps us to differentiate our custom metrics with default ones generated by our infra.